### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Galois): fiber functor is essentially surjective

### PR DESCRIPTION
**Pull request by chrisflav**
We show that every finite, discrete `Aut F`-set is in the essential image of the fiber functor.

---

- [x] depends on: #16478
- [x] depends on: #16441
- [x] depends on: #16668
- [x] depends on: #16948 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Galois/EssSurj.lean [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+open Action.FintypeCat
+
+variable (V : OpenSubgroup (Aut F)) {U : OpenSubgroup (Aut F)}
+  (h : Subgroup.Normal U.toSubgroup) {A : C} (u : (functorToAction F).obj A ≅ Aut F ⧸ₐ U.toSubgroup)
+
+/-
+
+### Strategy outline
+
+Let `A` be object of `C` with fiber `Aut F`-isomorphic to `Aut F ⧸ U` for an open normal
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Galois/EssSurj.lean:116]:
```suggestion
Let `A` be an object of `C` with fiber `Aut F`-isomorphic to `Aut F ⧸ U` for an open normal
```

File: Mathlib/CategoryTheory/Galois/EssSurj.lean [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+/-!
+
+# Essential surjectivity of fiber functors
+
+Let `F : C ⥤ FintypeCat` be a fiber functor of a Galois category `C` and denote by
+`H` the induced functor `C ⥤ Action FintypeCat (Aut F)`.
+
+In this file we show that the essential image of `H` are the finite `Aut F`-sets where
+the `Aut F` action is continuous.
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Galois/EssSurj.lean:18]:
```suggestion
In this file we show that the essential image of `H` consists of the finite `Aut F`-sets where
the `Aut F` action is continuous.
```

File: Mathlib/CategoryTheory/Galois/EssSurj.lean [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+- `exists_lift_of_continuous`: If `X` is a finite, discrete `Aut F`-set, then
+  there exists an object `A` such that `F.obj A` is isomorphic to `X` as
+  `Aut F`-sets.
+
+## Strategy
+
+We first show that every finite, discrete `Aut F`-set `Y` has a decomposition into connected
+components and each connected component is of the form `Aut F ⧸ U` for an open subgroup `U`.
+Since `H` preserves finite coproducts, it hence suffices to treat the case `Y = Aut F ⧸ U`.
+For the case `Y = Aut F ⧸ U` we closely follow the second part of Stacks Project Tag 0BN4.
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Galois/EssSurj.lean:34]: Should there be Stacks tags on some of the results below? Now that we have infrastructure for this...

Comment (chrisflav) [reply Mathlib/CategoryTheory/Galois/EssSurj.lean:34]: I had not noticed that we now have infrastructure for this, this is great! I should open a PR sometime adding a bunch of stacks tags in the Galois category files.

**Comment by jcommelin** [comment]
maintainer merge

File: Mathlib/CategoryTheory/Galois/EssSurj.lean [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+-/
+
```
Comment (kim-em) [inline Mathlib/CategoryTheory/Galois/EssSurj.lean:37]: Just write `noncomputable theory` somewhere up here instead of all the repeated `noncomputable`?

**Comment by kim-em** [comment]
bors d+

File: Mathlib/CategoryTheory/Galois/EssSurj.lean (hunk 4 above)
Comment (chrisflav) [reply Mathlib/CategoryTheory/Galois/EssSurj.lean:37]: `noncomputable theory` does not seem to work, but I added `noncomputable section`. I hope that's what you meant.

**Comment by chrisflav** [comment]
Thanks for the reviews!

bors r+

**Pull request closed without merging**